application-services/custom/influxdb-export/pkg/transforms/converions.go

```diff
@@ -0,0 +1,70 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Systems LTD
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package transforms
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/appcontext"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type Conversion struct {
+}
+
+func NewConversion() Conversion {
+	return Conversion{}
+}
+
+func (f Conversion) TransformToInflux(edgexcontext *appcontext.Context, params ...interface{}) (continuePipeline bool, stringType interface{}) {
+	if len(params) < 1 {
+		return false, errors.New("No Event Received")
+	}
+
+	edgexcontext.LoggingClient.Debug("Transforming to InfluxDB Line Protocol format")
+
+	event, ok := params[0].(models.Event)
+
+	if !ok {
+		return false, errors.New("Unexpected type received")
+	}
+
+	var buffer strings.Builder
+
+	// write device name as measurement
+	buffer.WriteString(event.Device)
+	// write tags if any, comma separated [,tag<index>=<tag_value>[,tag<index>=<tag_value>]]
+	for i, tag := range event.Tags {
```

Comment:
Problem I have here is with maps `i` isn't an index it is the key value. In our case it is the tag name.

So I expect this code here to generate the following which doesn't match the comment:
```
,tag<tag-name>=<tag-value>, tag<tag-name>=<tag-value>
```
See code here for example of looping through the tags:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/event.go#L94


Does the <tag-name> new the word `tag` prefixed to it?

Reply:
Yes - tag is prefixed to i.  Again, the index with the work tag becomes the key for tags.

// Syntax
<measurement>[,**<tag_key>**=<tag_value>[,**<tag_key>**=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>]

// Example
myMeasurement,**tag1**=value1,**tag2**=value2 fieldKey="fieldValue" 1556813561098000000